cmd/cli/proxy_configdump.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/pkg/action"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+const dumpConfigDescription = `
+This command will dump the sidecar proxy configuration for the given pod.
+`
+
+const dumpConfigExample = `
+# Dump the proxy configuration for pod 'bookbuyer-5ccf77f46d-rc5mg' in the 'bookbuyer' namespace
+osm proxy dump-config bookbuyer-5ccf77f46d-rc5mg -n bookbuyer
+`
+
+type proxyDumpConfigCmd struct {
+	out        io.Writer
+	config     *rest.Config
+	clientSet  kubernetes.Interface
+	namespace  string
+	pod        string
+	localPort  uint16
+	sigintChan chan os.Signal
+}
+
+func newProxyDumpConfig(config *action.Configuration, out io.Writer) *cobra.Command {
+	dumpConfigCmd := &proxyDumpConfigCmd{
+		out:        out,
+		sigintChan: make(chan os.Signal, 1),
+	}
+
+	cmd := &cobra.Command{
+		Use:   "dump-config POD ...",
+		Short: "dump proxy config",
+		Long:  dumpConfigDescription,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			dumpConfigCmd.pod = args[0]
+			conf, err := config.RESTClientGetter.ToRESTConfig()
+			if err != nil {
+				return errors.Errorf("Error fetching kubeconfig")
+			}
+			dumpConfigCmd.config = conf
+
+			clientset, err := kubernetes.NewForConfig(conf)
+			if err != nil {
+				return errors.Errorf("Could not access Kubernetes cluster. Check kubeconfig")
+			}
+			dumpConfigCmd.clientSet = clientset
+			return dumpConfigCmd.run()
+		},
+		Example: dumpConfigExample,
+	}
+
+	//add mesh name flag
+	f := cmd.Flags()
+	f.StringVarP(&dumpConfigCmd.namespace, "namespace", "n", metav1.NamespaceDefault, "Namespace of pod")
```

Comment:
There's already a `-n/--namespace` flag automatically added to every command from pkg/cli/environment.go to define the control plane namespace. Does it still work to define both the pod namespace and control plane namespace? If the control plane namespace doesn't need to be defined, then do both flags show up in the help text? That would be confusing.

Reply:
I would not mix the pod namespace with the control plane namespace. This is what the help text looks like:
```
This command will dump the sidecar proxy configuration for the given pod.

Usage:
  osm proxy dump-config POD ... [flags]

Examples:

# Dump the proxy configuration for pod bookbuyer-5ccf77f46d-rc5mg in the bookbuyer namespace
osm proxy dump-config bookbuyer-5ccf77f46d-rc5mg -n bookbuyer


Flags:
  -h, --help                help for dump-config
  -p, --local-port uint16   Local port to use for port forwarding (default 15000)

Global Flags:
  -n, --namespace string   namespace for osm control plane (default "osm-system")
```

Reply:
So then does the flag on the command work to change the pod namespace?

Reply:
The flag on the command is meant to specify the pod namespace, not to change it. The pod here is the application pod, not the controller pod. If you did specify a controller pod, the command will error due to the controller not being a `meshed pod`.

Reply:
I filed https://github.com/openservicemesh/osm/issues/1884 to track the ambiguity between the namespace flags.